src/index.js

```diff
@@ -13,34 +26,93 @@ module.exports = new BaseKonnector(start)
 // The start function is run by the BaseKonnector instance only when it got all the account
 // information (fields). When you run this connector yourself in "standalone" mode or "dev" mode,
 // the account information come from ./konnector-dev-config.json file
-function start(fields) {
+async function start(fields) {
+  await authenticate(fields.login, fields.password)
   // The BaseKonnector instance expects a Promise as return of the function
-  return request(`${baseUrl}/index.html`).then($ => {
-    // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
-    // here I do an Array.from to convert the cheerio fake array to a real js array.
-    const entries = Array.from($('article')).map(article =>
-      parseArticle($, article)
-    )
-    return addData(entries, 'io.cozy.books').then(() =>
-      saveFiles(entries, fields)
-    )
+  const $ = await request(`${baseUrl}/index.html`)
+  // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
+  const documents = await parseDocuments($)
+
+  // here we use the saveBills function even if what we fetch are not bills, but this is the most
+  // common case in connectors
+  await saveBills(documents, fields.folderPath, {
+    // this is a bank identifier which will be used to link bills to bank operations. These
+    // identifiers should be at least a word found in the title of a bank operation related to this
+    // bill. It is not case sensitive.
+    identifiers: ['books']
+  })
+}
+
+// this shows authentication using the [signin function](https://github.com/konnectors/libs/blob/master/packages/cozy-konnector-libs/docs/api.md#module_signin)
+// even if this in another domain here, but it works as an example
+function authenticate(username, password) {
+  return signin({
+    url: `http://quotes.toscrape.com/login`,
+    formSelector: 'form',
+    formData: { username, password },
+    // the validate function will check if
+    validate: (statusCode, $) => {
+      // The login in toscrape.com always works excepted when no password is set
+      if ($(`a[href='/logout']`).length === 1) {
+        return true
+      } else {
+        // cozy-konnector-libs has its own logging function which format these logs with colors in
+        // standalone and dev mode and as JSON in production mode
+        log('error', $('.error').text())
+        return false
+      }
```

Comment:
```javascript
if ($(`a[href='/logout']`).length === 1) return true

const $error = $('.error')
if (error) log('error', $error.text())
else log('debug', `Login failed for unknown reason: ${$.html()}`)
return false
```
or equivalent? (makes a good example of handling weird cases...)

Reply:
Also doesn't the `signIn()` function distinguish between login failures and other weird errors?